### DISCUSSION TITLE
Can't find a view activity related values?

### DISCUSSION THREAD

**Discussion by jessesavvy** [cyberjunky/python-garminconnect#198] [Q&A]
Hi all,

First, thanks for the awesome library.

I'm having difficulty finding a few activity related values in the API.

1. "Total Intensity". I have found "moderateIntensityMinutes" and "vigorousIntensityMinutes" but in the Garmin app some of my activities I see a "2X" next to my vigorous minutes. Is this in the API anywhere? Or is "Total Intensity" minutes?

2. I can't find the "Self Evaluation" data for an activity, i.e. "How Did You Feel?" and "Perceived Effort". 

3. And I'm looking for the "Time in Heart Rate Zones", i.e. number of minutes & seconds in each of Zones 1-5. 

If this is the wrong place to ask, apologies -- and thanks in advance regardless,

Jesse

**Comment by blakesims** [comment]
Hi Jesse, 
I am also looking for the same data in regards to the Time in heart rate zones. I can see that is shown quite nicely in the garmin connect app, so it must be accessible in the API. 
Did you end up working it out?

**Comment by blakesims** [comment]
After some digging around I found the required API endpoint. 
This did the trick for me:

```
    # Get activities for a date range
    start_date = "2024-06-01"
    end_date = "2024-06-13"
    activities = api.get_activities_by_date(start_date, end_date)
    print(f"Activities from {start_date} to {end_date}:")
    print(activities)
    print()

    # Get time in zone heart rate data per activity id 
    if activities:
        activity_id = activities[0]["activityId"]
        hr_timezones = api.get_activity_hr_in_timezones(activity_id)
        print(f"Time in HR zones for Activity ID {activity_id}:")
        print(hr_timezones)
        print()

```

I found this and all of the available API endpoints using this:

```
import inspect
import os

from garminconnect import Garmin


# Initialize the Garmin API
def init_api():
    email = os.environ.get("GARMIN_EMAIL")
    password = os.environ.get("GARMIN_PASSWORD")

    if not email or not password:
        print("Missing GARMIN_EMAIL or GARMIN_PASSWORD environment variables.")
        return None

    try:
        api = Garmin(email, password)
        api.login()
        return api
    except Exception as e:
        print(f"Failed to initialize Garmin API: {str(e)}")
        return None


def list_api_functions(api):
    functions = inspect.getmembers(api, predicate=inspect.ismethod)
    api_functions = [func for func in functions if not func[0].startswith("_")]
    return api_functions


def main():
    api = init_api()
    if api:
        api_functions = list_api_functions(api)
        print("Available API functions:")
        for func_name, func in api_functions:
            print(f"- {func_name}")
    else:
        print("Failed to initialize Garmin API.")


if __name__ == "__main__":
    main()
```
Hope that is useful.

**Comment by RobinVH** [comment]
I've also been searching how to get the RPEn (Perceived Effort) data in my tracker. It is buried in the json activitity data. You can get it per activity with: 

```
activity = api.get_activity(activity_id)
rpe = activity.get("summaryDTO", {}).get("directWorkoutRpe")
if rpe is not None:
    rpe = rpe / 10  # Convert to normal 1–10 scale
```

**Reply by tcibils** [reply]
Thank you for that comment, very helpful! Claude wasn't finding how to get that data and your comment fixed it 👍
